src/nvim/cursor_shape.c

```diff
@@ -99,11 +99,23 @@ char_u *parse_shape_opt(int what)
 
   // First round: check for errors; second round: do it for real.
   for (round = 1; round <= 2; round++) {
+    // Set the defaults, for the missing parts
+    if (round == 2 || *p_guicursor == NUL) {
+      for (idx = 0; idx < SHAPE_IDX_COUNT; idx++) {
```

Comment:
this for loop should go in a `static clear_shap_table()` function. That avoids setting `idx` here which is fragile, also helps readability.